### DISCUSSION TITLE
How can I include the base URI and custom headers in the API test cases using tcases

### DISCUSSION THREAD

**Discussion by soumyadipmitralpl** [Cornutum/tcases#191] [Q&A] [answered]
Hi,

While generating API test cases using the below command, I am seeing that the base URI is not getting included to the tests. My endpoints also need an authentication header (bearer token) to be added to each request.

> tcases-api-test -o src/test/java/org/examples *filename* 

How can I make tcases generate the API test cases with the baseURI() and headers() populated in it with my defined values? Thanks again.

![tcases_query1](https://user-images.githubusercontent.com/72627465/123468636-08064d80-d5c0-11eb-9048-ba031686c2d8.png)

**Comment by kerrykimbrough** [comment, marked as answer]
For the base URI, Tcases uses the first element of the `servers` property defined by the OpenAPI document. Your example shows what happens when `servers` is missing or empty. Although, when defined, you'll see this server URI included in the `request` URL, not in the `baseUri` setting.

It's reasonable to expect that the `servers` array could list different URIs for different test environments. In which case, it would be helpful to have a way to select the one to use for the generated test, using a parameter specified at runtime. Unfortunately, there's no well-defined identifier associated with a Server object.  One solution might be a parameter that specifies a regex to match the `server.description`. Or a parameter that specifies the server URI directly, which could be helpful when generating tests for an environment not listed in the OpenAPI doc. See issue #192.

**Reply by soumyadipmitralpl** [reply]
I think an option to define a parameter would be very helpful. Thank you for your reply.

**Comment by kerrykimbrough** [comment]
For the authentication token, this is currently not supported. See issue #115.

**Reply by soumyadipmitralpl** [reply]
Thanks. Is there any way to include a custom header key-value pairs in the tests generated? That might help too. Thanks.

**Reply by kerrykimbrough** [reply]
HTTP Bearer authentication is now supported in version 3.7.0. For details, see the [release notes](https://github.com/Cornutum/tcases/blob/master/ReleaseNotes.md#370).
